proteinflow/data/torch.py

```diff
@@ -27,20 +27,36 @@ def pad_collate(self, batch):
         # pad according to max_len
         to_pad = [max_len - b["S"].shape[0] for b in batch]
         for key in batch[0].keys():
-            if key in ["chain_id", "chain_dict", "pdb_id", "cdr_id"]:
+            if key in ["chain_id", "chain_dict", "pdb_id", "cdr_id", "ligand_smiles"]:
                 continue
-            out[key] = torch.stack(
-                [
-                    torch.cat([b[key], torch.zeros((pad, *b[key].shape[1:]))], 0)
-                    for b, pad in zip(batch, to_pad)
-                ],
-                0,
-            )
+            if key == "X_ligands" or key == "ligand_chains":
+                max_len_lig = max([b[key].shape[0] for b in batch])
```

Comment:
not super important but I'd move the max len computation to line 29 to make the code more readable (and to not recompute it multiple times)